src/compiler/checker.rs

```diff
@@ -468,6 +733,66 @@ impl TypeChecker {
         }
     }
 
+    fn convert_auto_to_any(&self, type_: Rc<Type>) -> Rc<Type> {
+        type_
+    }
+
+    fn check_variable_like_declaration(&mut self, node: &VariableDeclaration, wrapper: Rc<Node>) {
+        if true {
+            self.check_source_element(node.type_());
+        }
+
+        let symbol = self.get_symbol_of_node(node).unwrap();
+
+        let type_ = self.convert_auto_to_any(self.get_type_of_symbol(&*symbol));
+        let value_declaration = symbol.maybe_value_declaration();
+        if value_declaration.is_some()
+            && Rc::ptr_eq(
+                &wrapper,
+                &value_declaration.as_ref().unwrap().upgrade().unwrap(),
+            )
+        {
+            let initializer = get_effective_initializer(&*wrapper);
+            if let Some(initializer) = initializer {
+                if true {
+                    let initializer_type = self.check_expression_cached(match &*initializer {
+                        Node::Expression(expression) => expression,
+                        _ => panic!("Expected Expression"),
+                    });
+                    self.check_type_assignable_to_and_optionally_elaborate(
```

Comment:
This is where it's actually checking whether the declared type of the variable is compatible with the type of its initializer